tests/test_generichash.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2016 Donald Stufft and individual contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+
+import binascii
+import json
+import os
+
+import pytest
+
+import nacl.encoding
+import nacl.exceptions as exc
+import nacl.hash
+from nacl.utils import bytes_as_string
+
+OVERLONG_PARAMS_VECTORS = [
+    (b'key', 65*b'\xaa', 16*b'\xaa', 16*b'\x55', 64, b'will raise'),
+    (b'salt', b'key', 17*b'\xaa', 16*b'\x55', 64, b'will raise'),
+    (b'personal', b'key', 16*b'\xaa', 17*b'\x55', 64, b'will raise'),
+    (b'digest_size', b'key', 16*b'\xaa', 16*b'\x55', 65, b'will raise'),
+]
+
+
+def _read_crypto_test_vectors(fname):
+    vectors = []
+    path = os.path.join(os.path.dirname(__file__), "data", fname)
+    with open(path, "r") as fp:
```

Comment:
I've just tried to implement your suggestion, but, while all goes well on python2.7, all I've got on python3.5 is a crash, deep inside py.test during test collection, before the real test could even start:

```
==================================== ERRORS ====================================
__________________ ERROR collecting tests/test_generichash.py __________________
venv-35/lib/python3.5/site-packages/_pytest/runner.py:163: in __init__
    self.result = func()
venv-35/lib/python3.5/site-packages/_pytest/main.py:464: in _memocollect
    return self._memoizedcall('_collected', lambda: list(self.collect()))
venv-35/lib/python3.5/site-packages/_pytest/main.py:335: in _memoizedcall
    res = function()
# # # [  .  .  .  ]
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
venv-35/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py:614: in execute
    res = hook_impl.function(*args)
venv-35/lib/python3.5/site-packages/_pytest/python.py:110: in pytest_generate_tests
    metafunc.parametrize(*marker.args, **marker.kwargs)
venv-35/lib/python3.5/site-packages/_pytest/python.py:856: in parametrize
    assert len(valset) == len(argnames)
E   AssertionError
```

I think it's better to leave things as they stand now.